File: 백준/Silver/10845. 큐/큐.cc
#include <iostream>
#include <queue>
#include <string>

using namespace std;

template<typename T>
void command(queue<T> &q, string &str) {
    if (str == "push") {
        int x;
        cin >> x;
        q.push(x);
    }
    
    else if (str == "pop") {
        if(!q.empty()) {
            cout << q.front() << endl;
            q.pop();
        }
        
        else cout << -1 << endl;
    }
    
    else if (str == "size") {
        cout << q.size() << endl;
    }
    
    else if (str == "empty") {
        if(q.empty()) cout << 1 << endl;
        else cout << 0 << endl;
    }
    
    else if (str == "front") {
        if(!q.empty()) {
            cout << q.front() << endl;
        }
        else cout << -1 << endl;
    }
    
    else if (str == "back") {
        if(!q.empty()) {
            cout << q.back() << endl;
        }
        else cout << -1 << endl;
    }
    
    else;
}

int main(int argc, char** argv) {
    int N;
    queue<int> q;
    
    cin >> N;
    
    for(auto i = 0; i < N; i++) {
        string str;
        cin >> str;
        command<int>(q, str);
    }
    
    return 0;
}

File: 백준/Silver/10845. 큐/큐.py
import sys

N = int(sys.stdin.readline())
q = []

for i in range(N) :
    command = sys.stdin.readline().split()

    if command[0] == 'push' : q.append(command[1])

    elif command[0] == 'pop' : 
        print(q.pop(0)) if len(q) != 0 else print(-1)

    elif command[0] == 'size' : print(len(q))

    elif command[0] == 'empty' :
        print(1) if len(q) == 0 else print(0)
            
    elif command[0] == 'front' :
        print(-1) if len(q) == 0 else print(q[0])
    
    elif command[0] == 'back' :
        print(-1) if len(q) == 0 else print(q[-1])